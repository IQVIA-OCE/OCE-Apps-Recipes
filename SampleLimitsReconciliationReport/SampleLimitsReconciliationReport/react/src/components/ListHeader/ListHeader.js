import React from 'react';
import { Platform, StyleSheet, Text, View } from 'react-native';
import { useTheme } from '@oce-apps/apollo-react-native';
import color from 'color';

export const ListHeader = ({ titles }) => {
  const theme = useTheme();
  const titleTextColor = theme.dark ? theme.colors.text : theme.colors.surface;

  return (
    <View style={[styles.listHeader,
      {
        backgroundColor: color(theme.colors.secondary).lighten(0.001).hex(),
        borderBottomColor: theme.colors.surface,
      }]}>
      {titles.map((field, key) => (
        <View key={key} style={[styles.headerCell, { borderRightColor: theme.colors.surface }]}>
          <Text style={[styles.title, { color: titleTextColor }]}>{field}</Text>
        </View>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  listHeader: {
    flexDirection: 'row',
    borderBottomWidth: 4,
  },
  headerCell: {
    flex: 1,
    flexDirection: 'row',
    borderRightWidth: 2,
  },
  title: {
    fontSize: Platform.OS !== 'web' ? 18 : '',
    flex: 2,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 10,
    marginBottom: 10,
    fontWeight: 'bold',
  },
});
