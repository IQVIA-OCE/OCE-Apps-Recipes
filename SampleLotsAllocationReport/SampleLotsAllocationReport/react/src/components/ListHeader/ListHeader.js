import React, { useState } from 'react';
import { StyleSheet, View, TouchableOpacity } from 'react-native';
import Icon from "react-native-vector-icons/MaterialCommunityIcons";
import color from 'color'
import { neutral01, useTheme, Text } from 'apollo-react-native';

export const ListHeader = ({ titles, onSort }) => {
  const theme = useTheme();
  const [sortOption, setSortoption] = useState(null);
  const [sortBy, setSortBy] = useState(false);
  const onSortColumn = (obj) => {
    onSort(obj.sortField, !sortBy);
    setSortBy(!sortBy)
    setSortoption(obj.fieldName);
  }
  return (
    <View style={[styles.listHeader, {
      backgroundColor: theme.dark ? color(theme.colors.surface).lighten(0.5).hex() : neutral01[500],
      borderBottomColor: theme.dark ? color(theme.colors.surface).darken(0.5).hex() : 'white'
    }]}>
      {(titles).map((field, key) => (
        <TouchableOpacity disabled={!field.sortField} testID={`sortColumn_${key}`} key={key} style={[styles.headerCell, {
          borderRightColor: theme.dark ? color(theme.colors.surface).darken(0.5).hex() : 'white'
        }]} onPress={() => onSortColumn(field)}>
          <View style={styles.cell}>
            <Text style={styles.title}>{field.title}</Text>
            {(sortOption === field.fieldName) && <Icon testID={`sortIcon_${key}`} color={theme.colors.text} name={sortBy ? "arrow-down" : "arrow-up"} style={styles.icon} size={14} />}
          </View>
        </TouchableOpacity>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  listHeader: {
    flexDirection: 'row',
    marginTop: 10,
    borderBottomWidth: 4,
  },
  headerCell: {
    flex: 1,
    flexDirection: 'row',
    borderRightWidth: 2,
    alignContent: 'center',
    alignItems: 'center'
  },
  cell: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',

  },
  title: {
    fontSize: 16,
    paddingVertical: 10,
    paddingLeft: 10,
    fontWeight: 'bold',
  },
  icon: {
    paddingLeft: 4,
  }
});
