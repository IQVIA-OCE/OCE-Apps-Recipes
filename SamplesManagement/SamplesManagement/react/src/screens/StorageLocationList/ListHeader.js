import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { Colors } from 'apollo-react-native';

const ListHeader = () => {
  return (
    <View style={styles.root}>
      <View style={styles.col}>
        <Text style={styles.title}>ADDRESS</Text>
      </View>
      <View style={styles.col}>
        <Text style={styles.title}>DEFAULT STORAGE LOCATION</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    flexDirection: 'row',
    paddingVertical: 5,
    paddingHorizontal: 10,
    borderBottomWidth: 1,
    backgroundColor: Colors.themeGrey[50],
    borderColor: Colors.themeGrey[200],
  },
  col: {
    flex: 1,
  },
  title: {
    fontWeight: 'bold',
  },
});

export default ListHeader;
