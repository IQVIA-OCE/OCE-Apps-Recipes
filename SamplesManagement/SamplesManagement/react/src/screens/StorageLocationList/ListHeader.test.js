import React from 'react';
import renderer from 'react-test-renderer';
import ListHeader from './ListHeader';
import {IconButton} from "apollo-react-native";

describe('ListHeader', () => {
  it('Should render ListHeader component', () => {
    const tree = renderer.create(<ListHeader />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
