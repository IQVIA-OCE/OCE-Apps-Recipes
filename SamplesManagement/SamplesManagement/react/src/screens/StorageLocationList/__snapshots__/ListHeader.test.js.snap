// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader Should render ListHeader component 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      },
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#DADADA",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      ADDRESS
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      DEFAULT STORAGE LOCATION
    </Text>
  </View>
</View>
`;
